src/you_get/common.py

```diff
@@ -1056,216 +1055,211 @@ def download_main(download, download_playlist, urls, playlist, **kwargs):
+    if args.player:
+        player = args.player
+        caption = False
+
+    if args.no_proxy:
+        set_http_proxy('')
+    else:
+        set_http_proxy(args.http_proxy)
+    if args.socks_proxy:
+        set_socks_proxy(args.socks_proxy)
+
+    URLs = []
+    if args.input_file:
+        logging.debug('you are trying to load urls from %s', args.input_file)
+        if args.playlist:
+            log.e("reading playlist from a file is unsupported and won't make your life easier")
             sys.exit(2)
-    if not args and not urls_from_file:
-        print(help)
-        sys.exit()
-    args.extend(urls_from_file)
+        URLs.extend(args.input_file.read().splitlines())
+        args.input_file.close()
+    URLs.extend(args.URL)
 
-    if (socks_proxy):
-        try:
-            import socket
-            import socks
-            socks_proxy_addrs = socks_proxy.split(':')
-            socks.set_default_proxy(socks.SOCKS5,
-                                    socks_proxy_addrs[0],
-                                    int(socks_proxy_addrs[1]))
-            socket.socket = socks.socksocket
-            def getaddrinfo(*args):
-                return [(socket.AF_INET, socket.SOCK_STREAM, 6, '', (args[0], args[1]))]
-            socket.getaddrinfo = getaddrinfo
-        except ImportError:
-            log.w('Error importing PySocks library, socks proxy ignored.'
-                'In order to use use socks proxy, please install PySocks.')
-    else:
-        import socket
-        set_http_proxy(proxy)
+    if not URLs:
+        parser.print_help()
+        sys.exit()
 
-    socket.setdefaulttimeout(timeout)
+    socket.setdefaulttimeout(args.timeout)
 
     try:
+        extra = {}
+        if extractor_proxy:
+            extra['extractor_proxy'] = extractor_proxy
         if stream_id:
-            if not extractor_proxy:
-                download_main(download, download_playlist, args, playlist, stream_id=stream_id, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
-            else:
-                download_main(download, download_playlist, args, playlist, stream_id=stream_id, extractor_proxy=extractor_proxy, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
-        else:
-            if not extractor_proxy:
-                download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
-            else:
-                download_main(download, download_playlist, args, playlist, extractor_proxy=extractor_proxy, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
+            extra['stream_id'] = stream_id
+        download_main(
+            download, download_playlist,
+            URLs, args.playlist,
+            stream_id=stream_id, output_dir=args.output_dir, merge=not args.no_merge,
```

Comment:
Be careful not to duplicate the keyword argument `stream_id` here. (it's already in `extra`)

Reply:
Oh yes, thanks.